purchase_order_supplierinfo_update/models/purchase_order.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2021 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def write(self, vals):
+        no_updated = self.filtered(lambda r: r.state not in ["purchase", "done"])
+        res = super().write(vals)
+        if vals.get("state", "") in ["purchase", "done"]:
+            no_updated.mapped("order_line").update_supplierinfo_price()
+        return res
+
+
+class PurchaseOrderLine(models.Model):
```

Comment:
Please split one model per file.

Reply:
We can consider the Lines as intimately connected to the Header document, and functionally effectively the same thing.
So I believe we are better keeping them on the same file, like the original author did.